akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala

```diff
@@ -149,8 +152,21 @@ private object PoolSlot {
       case FromConnection(OnNext(response: HttpResponse)) ⇒
         val requestContext = inflightRequests.head
         inflightRequests = inflightRequests.tail
-        val delivery = ResponseDelivery(ResponseContext(requestContext, Success(response)))
-        val requestCompleted = SlotEvent.RequestCompleted(slotIx)
+        val (entity, whenCompleted) = response.entity match {
+          case x: HttpEntity.Strict ⇒ x -> FastFuture.successful(())
+          case x: HttpEntity.Default ⇒
+            val (newData, whenCompleted) = StreamUtils.captureTermination(x.data)
+            x.copy(data = newData) -> whenCompleted
+          case x: HttpEntity.CloseDelimited ⇒
+            val (newData, whenCompleted) = StreamUtils.captureTermination(x.data)
+            x.copy(data = newData) -> whenCompleted
+          case x: HttpEntity.Chunked ⇒
+            val (newChunks, whenCompleted) = StreamUtils.captureTermination(x.chunks)
+            x.copy(chunks = newChunks) -> whenCompleted
```

Comment:
Actually its five implementations, `Chunked` was missing from my list above.
We definitely need this matrix.... :)